Frogies-Project/Assets/Scripts/Core/PotionSystem/PotionSystem.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using Items.Core;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Core.PotionSystem
+{
+    public class PotionSystem : MonoBehaviour
+    {
+        [SerializeField] private GameObject optionPrefab;
+        [SerializeField] private Transform body;
+        private List<PotionOption> _options;
+        public event EventHandler OnOptionSelected;
```

Comment:
Imho better to use Action(generic wrapper for EventHandler) but if you like it more this way you can leave it like that